astropy/io/misc/tests/test_hdf5.py

```diff
@@ -567,6 +567,45 @@ def test_read_h5py_objects(tmpdir):
     f.close()         # don't raise an error in 'test --open-files'
 
 
+@pytest.mark.skipif('not HAS_H5PY')
+def test_write_to_hdf5(tmpdir):
+    test_file = str(tmpdir.join('test.hdf5'))
+
+    t = Table()
+    t['p'] = ['a', 'b', 'c']
+    t['q'] = [ 1, 2, 3 ]
+    t['r'] = [b'a', b'b', b'c']
+    t['s'] = ["\u2119", "\u01b4", "\u2602"]
+    t.write(test_file, path='the_table', overwrite=True)
+    assert np.all(t['p'] == ['a', 'b', 'c'])
+    assert np.all(t['q'] == [1, 2, 3])
+    assert np.all(t['r'] == [b'a', b'b', b'c'])
+    assert np.all(t['s'] == ["\u2119", "\u01b4", "\u2602"])
```

Comment:
Are these four asserts really needed? They are just checking the original table. I think you can remove them